### PR TITLE
Milestone #6

### PR DESCRIPTION
**Pull request by BregornOriginal**
# The blog now can accept the input in form of comments and posts! :dizzy:

### Also was added a button like, that adds a like to the post

- Create a method called current_user in your ApplicationController to make current user data available to all controllers.
     - It will return the first user from the database.

- Create forms to perform the following functions:
     - Creates a Post on behalf of the current_user (use the method that you created in your ApplicationController).
- Create a comment on behalf of the current_user (use the method that you created in your ApplicationController).

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,5 @@
 class ApplicationController < ActionController::Base
+  def current_user
+    User.first
+  end
```
Comment (mateo951) [inline app/controllers/application_controller.rb:4]: ApplicationController contains #current_user method 🔥

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -1,12 +1,35 @@
...
+        if post.save
+          flash[:success] = 'Post saved successfully'
+          redirect_to users_url
+        else
+          flash.now[:error] = 'Error: Post could not be saved'
+          render :new, status: 422
+        end
+      end
+    end
+  end
```
Comment (mateo951) [inline app/controllers/posts_controller.rb:34]: Posts controller is perfect ✔

File: config/routes.rb [hunk 3]
```diff
@@ -1,7 +1,10 @@
 Rails.application.routes.draw do
   root 'users#index'
   resources :users, only: [:index, :show] do
-    resources :posts, only: [:index, :show]
+    resources :posts, only: [:index, :new, :create, :show] do
```
Comment (mateo951) [inline config/routes.rb:4]: There is a pair of routes in the config/routes.rb file for creating posts 😎

File: app/views/posts/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  </div>
+
+  <div>
+    <%= f.label :text %><br>
+    <%= f.text_field :text %>
+  </div>
+
+  <%= f.submit "Create Post" %>
+
+<% end %>
```
Comment (mateo951) [inline app/views/posts/new.html.erb:18]: There is a form in the views/posts/new.html.erb file which accepts input for creating posts 🤓

File: app/controllers/likes_controller.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+      format.html do
+        if like.save
+          flash[:success] = 'You liked this post!'
+          redirect_to user_post_url(author, post)
+        else
+          flash.now[:error] = 'Error: Post could not be liked.'
+        end
+      end
+    end
+  end
```
Comment (mateo951) [inline app/controllers/likes_controller.rb:19]: A user should be able to add a like to posts ✔

**Review by mateo951 — APPROVED** [review]
### Hi @BregornOriginal, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🐐 
### Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 2 above)
Comment (BregornOriginal) [reply app/controllers/posts_controller.rb:34]: Hi! @mateo951 Thanks for your review and your appreciation!

File: app/controllers/likes_controller.rb (hunk 5 above)
Comment (BregornOriginal) [reply app/controllers/likes_controller.rb:19]: Thanks for the review @mateo951

**Pull request merged**